src/miner.cpp

```diff
@@ -573,9 +551,14 @@ void GenerateBitcoins(bool fGenerate, CWallet* pwallet, int nThreads)
     if (nThreads == 0 || !fGenerate)
         return;
 
+    CScript coinbaseScript;
+    GetMainSignals().ScriptForMining(coinbaseScript);
```

Comment:
What would asynchronous even mean in this context? You can specify how signal's return values are combined. There is no way to make the execution asynchronous without breaking the API (which returns the combined return value directly).